crates/telio-firewall/src/firewall.rs

```diff
@@ -610,6 +656,20 @@ impl StatefullFirewall {
     fn handle_inbound_icmp<'a, P: IpPacket<'a>>(&self, peer: &PublicKey, ip: &P) -> bool {
         let icmp_packet = unwrap_option_or_return!(P::Icmp::new(ip.payload()), false);
 
+        let key = IcmpConn {
+            remote_addr: ip.get_source().into(),
+            local_addr: ip.get_destination().into(),
+            identifier: 0,
+        };
+        let mut icmp_cache = unwrap_lock_or_return!(self.icmp.lock(), false);
+        if icmp_cache.get(&key).is_some() {
+            telio_log_trace!("Matched ICMP conntrack entry {:?}", key,);
+            telio_log_trace!("Removing ICMP conntrack entry {:?}", key);
+            icmp_cache.remove(&key);
+            telio_log_trace!("Accepting ICMP packet {:?} {:?}", ip, peer);
+            return true;
+        }
+
         if P::Icmp::BLOCKED_TYPES.contains(&icmp_packet.get_type()) {
```

Comment:
And maybe we want to think about letting BLOCKED_TYPES through in case of whitelisted peers? Not sure about that one. Just opening a discussion

Reply:
Never mind. I can see now that the whitelisting by peer is done earlier before this function